tests/test_iotedgedev_iothub_deploy.py

```diff
@@ -113,3 +114,82 @@ def test_iothub_deploy_error_missing_target_condition():
 
     # Assert
     assert "ERROR: Environment Variable IOTHUB_DEPLOYMENT_TARGET_CONDITION not set. Either add to .env file or to your system's Environment Variables" in str(context.value)
+
+
+def test_iothub_deploy_and_add_tags():
+    # Arrange
+    tags = '{"environment":"dev","building":"9"}'
+    deployment_name = f'test-{uuid.uuid4()}'
+    os.chdir(test_solution_shared_lib_dir)
+
+    # Act
+    result = runner_invoke(['build', '-f', "layered_deployment.flattened_props.template.json", '-P', get_platform_type()])
+    result = runner_invoke(['iothub', 'deploy',
+                            '-f', 'config/layered_deployment.flattened_props.json',
+                           '-n', deployment_name,
+                            '-p', '10',
+                            '-t', "tags.environment='dev'",
+                            '-dt', tags
+                            ])
+
+    # Assert
+    assert 'DEPLOYMENT COMPLETE' in result.output
+    assert 'TAG UPDATE COMPLETE' in result.output
+    assert tags in result.output
+    assert 'ERROR' not in result.output
+
+    azure_cli = AzureCli(output, envvars)
+
+    assert azure_cli.invoke_az_cli_outproc(["iot", "edge", "deployment", "delete", "-d", deployment_name, "-l", envvars.get_envvar("IOTHUB_CONNECTION_STRING")])
+    assert azure_cli.invoke_az_cli_outproc(["iot", "hub", "device-twin", "replace", "-d", DeviceConnectionString(envvars.get_envvar("DEVICE_CONNECTION_STRING")).device_id,
+                                            "-l", envvars.get_envvar("IOTHUB_CONNECTION_STRING"), "--json", "tag_overwrite.json"])
+
+
+def test_iothub_deploy_and_add_tags_retry_after_invalid_tag():
+    # GIVEN: `iothub deploy` run fails
+    # WHEN: Device tag has an invalid format
+    # THAN: Rerunning the command will lead to a failed deploy part,
```

Comment:
![image](https://user-images.githubusercontent.com/12761692/131978331-b77bcb7a-a0db-49e5-ad7c-cd14ef1876b9.png)
```suggestion
    # THEN: Rerunning the command will lead to a failed deploy part,
```

Reply:
Oh boy :rofl: 
Fixed now